packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+	DataObject,
+	DataObjectFactory,
+	IDataObjectProps,
+} from "@fluidframework/aqueduct";
+import { IContainer, LoaderHeader } from "@fluidframework/container-definitions";
+import { IFluidHandle, IRequest, IRequestHeader } from "@fluidframework/core-interfaces";
+import { SharedCounter } from "@fluidframework/counter";
+import { IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { IContainerRuntimeBase, IFluidDataStoreFactory } from "@fluidframework/runtime-definitions";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { SharedString } from "@fluidframework/sequence";
+import {
+	createAndAttachContainer,
+	createLoader,
+	createDocumentId,
+	LoaderContainerTracker,
+	ITestObjectProvider,
+} from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluid-internal/test-version-utils";
+import { IResolvedUrl } from "@fluidframework/driver-definitions";
+import { ContainerRuntime, ISummarizer, Summarizer } from "@fluidframework/container-runtime";
+
+const counterKey = "count";
+
+/**
+ * Implementation of counter dataObject for testing.
+ */
+export class TestDataObject extends DataObject {
+	public static readonly type = "@fluid-example/test-dataObject";
+
+	public static getFactory() {
+		return TestDataObject.factory;
+	}
+
+	private static readonly factory = new DataObjectFactory(
+		TestDataObject.type,
+		TestDataObject,
+		[],
+		{},
+	);
+
+	private counter!: SharedCounter;
+
+	/**
+	 * Expose the runtime for testing purposes.
+	 */
+
+	public runtime: IFluidDataStoreRuntime;
+
+	public constructor(props: IDataObjectProps) {
+		super(props);
+		this.runtime = props.runtime;
+	}
+
+	/**
+	 * Gets the current counter value.
+	 */
+	public get value(): number {
+		return this.counter.value;
+	}
+
+	/**
+	 * Increments the counter value by 1.
+	 */
+	public increment() {
+		this.counter.increment(1);
+	}
+
+	protected async initializingFirstTime() {
+		const counter = SharedCounter.create(this.runtime);
+		this.root.set(counterKey, counter.handle);
+	}
+
+	protected async hasInitialized() {
+		const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
+		assert(counterHandle);
+		this.counter = await counterHandle.get();
+	}
+}
+
+const testDataObjectFactory = new DataObjectFactory(
+	TestDataObject.type,
+	TestDataObject,
+	[SharedCounter.getFactory(), SharedString.getFactory()],
+	{},
+);
+
+// REVIEW: enable compat testing?
+describeNoCompat("LoadModes", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	before(() => {
+		provider = getTestObjectProvider();
+	});
+
+	const loaderContainerTracker = new LoaderContainerTracker();
+
+	let documentId: string;
+	let container1: IContainer;
+	let dataObject1: TestDataObject;
+
+	beforeEach(async () => {
+		documentId = createDocumentId();
+		container1 = await createContainer();
+		dataObject1 = await requestFluidObject<TestDataObject>(container1, "default");
```

Comment:
Since you're using ContainerRuntimeFactoryWithDefaultDataStore, probably prefer to use getDefaultObjectFromContainer instead (which is designed to pair with it).  Here and 2 more instances below

```suggestion
		dataObject1 = await getDefaultObjectFromContainer<TestDataObject>(container1);
```